diff --git a/node_modules/typescript/lib/_tsc.js b/node_modules/typescript/lib/_tsc.js
index 43be2b1..08179a4 100644
--- a/node_modules/typescript/lib/_tsc.js
+++ b/node_modules/typescript/lib/_tsc.js
@@ -7548,6 +7548,7 @@ var Diagnostics = {
   Compiles_the_current_project_with_additional_settings: diag(6929, 3 /* Message */, "Compiles_the_current_project_with_additional_settings_6929", "Compiles the current project, with additional settings."),
   true_for_ES2022_and_above_including_ESNext: diag(6930, 3 /* Message */, "true_for_ES2022_and_above_including_ESNext_6930", "`true` for ES2022 and above, including ESNext."),
   List_of_file_name_suffixes_to_search_when_resolving_a_module: diag(6931, 1 /* Error */, "List_of_file_name_suffixes_to_search_when_resolving_a_module_6931", "List of file name suffixes to search when resolving a module."),
+  Import_JSON_files_as_const_assertions: diag(6933, 3 /* Message */, "Import_JSON_files_as_const_assertions_6933", "Import JSON files as const assertions."),
   false_unless_checkJs_is_set: diag(6932, 3 /* Message */, "false_unless_checkJs_is_set_6932", "`false`, unless `checkJs` is set"),
   Variable_0_implicitly_has_an_1_type: diag(7005, 1 /* Error */, "Variable_0_implicitly_has_an_1_type_7005", "Variable '{0}' implicitly has an '{1}' type."),
   Parameter_0_implicitly_has_an_1_type: diag(7006, 1 /* Error */, "Parameter_0_implicitly_has_an_1_type_7006", "Parameter '{0}' implicitly has an '{1}' type."),
@@ -37603,6 +37604,15 @@ var commandOptionsWithoutBuild = [
     description: Diagnostics.Enable_importing_json_files,
     defaultValueDescription: false
   },
+  {
+    name: "importJsonAsConst",
+    type: "boolean",
+    affectsSemanticDiagnostics: true,
+    affectsBuildInfo: true,
+    category: Diagnostics.Language_and_Environment,
+    description: Diagnostics.Import_JSON_files_as_const_assertions,
+    defaultValueDescription: false
+  },
   {
     name: "allowArbitraryExtensions",
     type: "boolean",
@@ -56663,6 +56673,9 @@ function createTypeChecker(host) {
       if (!declaration.statements.length) {
         return emptyObjectType;
       }
+      if (compilerOptions.importJsonAsConst) {
+        return checkExpression(declaration.statements[0].expression);
+      }
       return getWidenedType(getWidenedLiteralType(checkExpression(declaration.statements[0].expression)));
     }
     if (isAccessor(declaration)) {
@@ -80717,6 +80730,9 @@ function createTypeChecker(host) {
   }
   function isConstContext(node) {
     const parent = node.parent;
+    if (compilerOptions.importJsonAsConst && parent.kind === 245 /* ExpressionStatement */ && isSourceFile(parent.parent) && isJsonSourceFile(parent.parent)) {
+      return true;
+    }
     return isAssertionExpression(parent) && isConstTypeReference(parent.type) || isJSDocTypeAssertion(parent) && isConstTypeReference(getJSDocTypeAssertionType(parent)) || isValidConstAssertionArgument(node) && isConstTypeVariable(getContextualType(node, 0 /* None */)) || (isParenthesizedExpression(parent) || isArrayLiteralExpression(parent) || isSpreadElement(parent)) && isConstContext(parent) || (isPropertyAssignment(parent) || isShorthandPropertyAssignment(parent) || isTemplateSpan(parent)) && isConstContext(parent.parent);
   }
   function checkExpressionForMutableLocation(node, checkMode, forceTuple) {
diff --git a/node_modules/typescript/lib/typescript.d.ts b/node_modules/typescript/lib/typescript.d.ts
index 0f39eae..aaf00bc 100644
--- a/node_modules/typescript/lib/typescript.d.ts
+++ b/node_modules/typescript/lib/typescript.d.ts
@@ -7043,6 +7043,7 @@ declare namespace ts {
         forceConsistentCasingInFileNames?: boolean;
         ignoreDeprecations?: string;
         importHelpers?: boolean;
+        importJsonAsConst?: boolean;
         /** @deprecated */
         importsNotUsedAsValues?: ImportsNotUsedAsValues;
         inlineSourceMap?: boolean;
diff --git a/node_modules/typescript/lib/typescript.js b/node_modules/typescript/lib/typescript.js
index d322bce..5c52000 100644
--- a/node_modules/typescript/lib/typescript.js
+++ b/node_modules/typescript/lib/typescript.js
@@ -10954,6 +10954,7 @@ var Diagnostics = {
   Compiles_the_current_project_with_additional_settings: diag(6929, 3 /* Message */, "Compiles_the_current_project_with_additional_settings_6929", "Compiles the current project, with additional settings."),
   true_for_ES2022_and_above_including_ESNext: diag(6930, 3 /* Message */, "true_for_ES2022_and_above_including_ESNext_6930", "`true` for ES2022 and above, including ESNext."),
   List_of_file_name_suffixes_to_search_when_resolving_a_module: diag(6931, 1 /* Error */, "List_of_file_name_suffixes_to_search_when_resolving_a_module_6931", "List of file name suffixes to search when resolving a module."),
+  Import_JSON_files_as_const_assertions: diag(6933, 3 /* Message */, "Import_JSON_files_as_const_assertions_6933", "Import JSON files as const assertions."),
   false_unless_checkJs_is_set: diag(6932, 3 /* Message */, "false_unless_checkJs_is_set_6932", "`false`, unless `checkJs` is set"),
   Variable_0_implicitly_has_an_1_type: diag(7005, 1 /* Error */, "Variable_0_implicitly_has_an_1_type_7005", "Variable '{0}' implicitly has an '{1}' type."),
   Parameter_0_implicitly_has_an_1_type: diag(7006, 1 /* Error */, "Parameter_0_implicitly_has_an_1_type_7006", "Parameter '{0}' implicitly has an '{1}' type."),
@@ -41886,6 +41887,15 @@ var commandOptionsWithoutBuild = [
     description: Diagnostics.Enable_importing_json_files,
     defaultValueDescription: false
   },
+  {
+    name: "importJsonAsConst",
+    type: "boolean",
+    affectsSemanticDiagnostics: true,
+    affectsBuildInfo: true,
+    category: Diagnostics.Language_and_Environment,
+    description: Diagnostics.Import_JSON_files_as_const_assertions,
+    defaultValueDescription: false
+  },
   {
     name: "allowArbitraryExtensions",
     type: "boolean",
@@ -61298,6 +61308,9 @@ function createTypeChecker(host) {
       if (!declaration.statements.length) {
         return emptyObjectType;
       }
+      if (compilerOptions.importJsonAsConst) {
+        return checkExpression(declaration.statements[0].expression);
+      }
       return getWidenedType(getWidenedLiteralType(checkExpression(declaration.statements[0].expression)));
     }
     if (isAccessor(declaration)) {
@@ -85352,6 +85365,9 @@ function createTypeChecker(host) {
   }
   function isConstContext(node) {
     const parent2 = node.parent;
+    if (compilerOptions.importJsonAsConst && parent2.kind === 245 /* ExpressionStatement */ && isSourceFile(parent2.parent) && isJsonSourceFile(parent2.parent)) {
+      return true;
+    }
     return isAssertionExpression(parent2) && isConstTypeReference(parent2.type) || isJSDocTypeAssertion(parent2) && isConstTypeReference(getJSDocTypeAssertionType(parent2)) || isValidConstAssertionArgument(node) && isConstTypeVariable(getContextualType2(node, 0 /* None */)) || (isParenthesizedExpression(parent2) || isArrayLiteralExpression(parent2) || isSpreadElement(parent2)) && isConstContext(parent2) || (isPropertyAssignment(parent2) || isShorthandPropertyAssignment(parent2) || isTemplateSpan(parent2)) && isConstContext(parent2.parent);
   }
   function checkExpressionForMutableLocation(node, checkMode, forceTuple) {
